### PR TITLE
Add generic felix config option

### PR DESCRIPTION
**Pull request by robbrockbank**
This requires a corresponding libcalico-go change to be merged.

### CODE REVIEW

File: calicoctl/commands/config.go [hunk 1]
```diff
@@ -317,3 +337,38 @@ func (a asnum) get(node string) error {
 	fmt.Println(asn.String())
 	return nil
 }
+
+// felixconfig implements the configType interface.
+type felixconfig struct {
```
Comment (fasaxc) [inline calicoctl/commands/config.go]: Should be `felixConfig`?

File: calicoctl/commands/config.go [hunk 2]
```diff
@@ -50,6 +56,10 @@ Examples:
 
 Options:
   -n --node=<NODE>      The node name.
+     --raw=<COMPONENT>  Apply raw configuration for the specified component.
+                        This option should generally not be used unless
+                        specifically required to set some low-level system
+                        configuration options.
```
Comment (fasaxc) [inline calicoctl/commands/config.go]: Should document the allowed value for `<COMPONENT>`

File: calicoctl/commands/config.go [hunk 3]
```diff
@@ -317,3 +337,38 @@ func (a asnum) get(node string) error {
...
+
+// felixconfig implements the configType interface.
+type felixconfig struct {
+	name string
+	c client.ConfigInterface
+}
+
+func (f felixconfig) set(value, node string) error {
+	if node != "" {
+		return errors.New("--node should not be specified")
```
Comment (fasaxc) [inline calicoctl/commands/config.go]: Why can't you specify a node?  That can be really useful to turn on logging for one host only

**Comment by fasaxc** [comment]
LGTM; please can you make sure they get some automated ST at some point?

**Comment by robbrockbank** [comment]
Yup - this was on my radar.   Automated STs are coming.  I will add once we get the STs PR merged - it's been blocked by a calico node fix that is taking it's time to materialize.  I'll raise an issue so I don't lose track of it.

**Comment by robbrockbank** [comment]
Issue #1330 raised

**Pull request merged**